### PR TITLE
Sanitize issue comment WS

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

**Comment by teryk** [comment]
@gregaubert please review https://github.com/SonarSource/sonarqube/pull/2087/commits/17ae1b6516ef9501ef1fde78b3fec98edb1300e2

File: server/sonar-server/src/main/resources/org/sonar/server/issue/ws/assign-example.json [hunk 1]
```diff
@@ -53,19 +53,15 @@
   },
   "components": [
     {
-      "id": 87163,
```
Comment (julienlancelot) [inline server/sonar-server/src/main/resources/org/sonar/server/issue/ws/assign-example.json]: A changelog entry should be added in api/issues/changelog WS

File: server/sonar-server/src/main/resources/org/sonar/server/issue/ws/do_transition-example.json [hunk 2]
```diff
@@ -53,19 +53,15 @@
   },
   "components": [
     {
-      "id": 87163,
```
Comment (julienlancelot) [inline server/sonar-server/src/main/resources/org/sonar/server/issue/ws/do_transition-example.json]: A changelog entry should be added in api/issues/do_transition WS

File: server/sonar-server/src/main/resources/org/sonar/server/issue/ws/set_severity-example.json [hunk 3]
```diff
@@ -53,19 +53,15 @@
   },
   "components": [
     {
-      "id": 87163,
```
Comment (julienlancelot) [inline server/sonar-server/src/main/resources/org/sonar/server/issue/ws/set_severity-example.json]: A changelog entry should be added in api/issues/set_severity WS

File: server/sonar-server/src/main/resources/org/sonar/server/issue/ws/set_type-example.json [hunk 4]
```diff
@@ -53,19 +53,15 @@
   },
   "components": [
     {
-      "id": 87163,
```
Comment (julienlancelot) [inline server/sonar-server/src/main/resources/org/sonar/server/issue/ws/set_type-example.json]: A changelog entry should be added in api/issues/set_type WS

File: sonar-ws/src/main/protobuf/ws-issues.proto [hunk 5]
```diff
@@ -65,7 +65,7 @@ message Issue {
   optional string rule = 2;
   optional sonarqube.ws.commons.Severity severity = 3;
   optional string component = 4;
-  optional int64 componentId = 5;
+  optional int64 deprecatedComponentId = 5;
```
Comment (julienlancelot) [inline sonar-ws/src/main/protobuf/ws-issues.proto]: As these fields are not deprecated but removed, we should use the other pattern "unusedXXX" (see in ws-rules.proto for example)

File: server/sonar-server/src/main/java/org/sonar/server/issue/ws/EditCommentAction.java [hunk 6]
```diff
@@ -66,10 +67,12 @@ public EditCommentAction(System2 system2, UserSession userSession, DbClient dbCl
...
     WebService.NewAction action = context.createAction(ACTION_EDIT_COMMENT)
       .setDescription("Edit a comment.<br/>" +
-        "Requires authentication and the following permission: 'Browse' on the project of the specified issue.<br/>" +
-        "Since 6.3, the response contains the issue with all details, not only the edited comment.<br/>" +
-        "Since 6.3, 'key' parameter has been renamed %s", PARAM_COMMENT)
+        "Requires authentication and the following permission: 'Browse' on the project of the specified issue.")
       .setSince("3.6")
+      .setChangelog(
+        new Change("6.3", "the response returns the issue with all its details"),
+        new Change("6.3", format("the 'key' parameter has been renamed %s", PARAM_COMMENT))
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/issue/ws/EditCommentAction.java]: typo : renamed **to** %s

Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/issue/ws/EditCommentAction.java]: You have forgotten this comment

File: sonar-ws/src/main/protobuf/ws-issues.proto (hunk 5 above)
Comment (julienlancelot) [inline sonar-ws/src/main/protobuf/ws-issues.proto]: Renaming this parameter was a good idea, but the name should start with "unused"

**Comment by julienlancelot** [comment]
2 feedbacks were not applied : 
- Typo in edit_comment changelog
- Renaming of componentId to unusedComponentId in ws-issues.proto

**Comment by teryk** [comment]
feedbacks were applied contrary of what you said:
* https://github.com/SonarSource/sonarqube/pull/2087/files#diff-7daabb240e8a1c0f85f23a59ad79cccfR68
* to rename doesn't need the to

**Comment by julienlancelot** [comment]
@teryk Oups, indeed I was wrong, I was looking into the wrong branch :(

**Pull request merged**